src/Components/Server/src/Circuits/CircuitHost.cs

```diff
@@ -136,6 +139,88 @@ internal void SendPendingBatches()
             var _ = Renderer.Dispatcher.InvokeAsync(() => Renderer.ProcessBufferedRenderBatches());
         }
 
+        internal async Task EndInvokeDotNetFromJS(string arguments)
+        {
+            try
+            {
+                AssertInitialized();
+
+                await Renderer.Dispatcher.InvokeAsync(() =>
+                {
+                    SetCurrentCircuitHost(this);
+                    var document = JsonDocument.Parse(arguments);
+                    if (document.RootElement.ValueKind != JsonValueKind.Array)
+                    {
+                        // Log error and return
+                        return;
+                    }
+                    var length = document.RootElement.GetArrayLength();
+                    if (length != 3)
+                    {
+                        // Log error and return
+                        return;
+                    }
+                    long asyncHandle = 0;
+                    bool succeeded = false;
+                    JSAsyncCallResult result = null;
+                    var i = 0;
+                    foreach (var element in document.RootElement.EnumerateArray())
+                    {
+                        if (i == 0)
+                        {
+                            asyncHandle = element.GetInt64();
+                        }
+                        if (i == 1)
+                        {
+                            succeeded = element.GetBoolean();
+                        }
+
+                        if (i == 2 && element.ValueKind != JsonValueKind.Null)
+                        {
+                            result = (JSAsyncCallResult)typeof(JSAsyncCallResult).GetConstructor(
+                                BindingFlags.NonPublic | BindingFlags.Instance,
+                                binder: null,
+                                new Type[] { typeof(JsonDocument), typeof(JsonElement) },
+                                null).Invoke(new object[] { document, element });
```

Comment:
Need to do this because the type cannot be publicly constructed, we will have to find a solution for this.